### ISSUE TITLE
HTTPResponse strict attribute gone in urllib3>=2.0.0

### ISSUE THREAD

**Issue by MadameMinty** [firebase/firebase-admin-python#699]
* Operating System version: Debian 11
  * Firebase SDK version: 6.1.0
  * Firebase Product: auth
  * Python version: 3.11.3
  * Pip version: 23.1.2

#### Steps to reproduce:

`firebase-admin==6.1.0` package requires `urllib3<3,>=1.21.1` and so pip installs `urllib3==2.0.2`, but with `urllib3>=2.0.0`, which removed attribute 'strict' from 'HTTPResponse' object, `firebase_admin.auth.verify_id_token(token)` raises an error `AttributeError: 'HTTPResponse' object has no attribute 'strict'`.

#### Relevant Code:

```python
import firebase_admin
from firebase_admin import auth
cred = firebase_admin.credentials.Certificate(GOOGLE_APPLICATION_CREDENTIALS)
firebase_admin.initialize_app(cred)
firebase_admin.auth.verify_id_token(token)
```

```
...
File "\Lib\site-packages\cachecontrol\serialize.py", line 54, in dumps
    u"strict": response.strict,
               ^^^^^^^^^^^^^^^
AttributeError: 'HTTPResponse' object has no attribute 'strict'
```

**Comment by Tom3652** [comment]
I have forced `urllib3==1.26.15` in the meantime.

**Comment by JefersonParraVeredata** [comment]
Thanks, urllib3 to 1.26.15 works well

**Comment by joente** [comment]
Any updates on this issue? (We use `urllib3<2.0.0`) as a workaround)

**Comment by jonathanedey** [comment]
This seems to have been an issue in the cachecontrol library and was tracked in https://github.com/psf/cachecontrol/issues/292. It was resolved in https://github.com/psf/cachecontrol/pull/301 and released in `cachecontrol >= 0.12.13`. We'll need to bump the minimum version to match that as well.
